lib/Test/Stream/Formatter/TAP.pm

```diff
@@ -99,6 +116,193 @@ sub _autoflush {
     select $old_fh;
 }
 
+sub event_tap {
+    my $self = shift;
+    my ($e, $num) = @_;
+
+    my $converter = $CONVERTERS{ref($e)};
+
+    $num = undef if $self->{+NO_NUMBERS};
+
+    # Legacy Support for $e->to_tap
+    unless ($converter) {
+        my $legacy = $e->can('to_tap') or return;
+        return if $legacy == \&Test::Stream::Event::to_tap;
+        warn "'$e' implements 'to_tap'. to_tap methods on events are deprecated.\n";
+        return $e->to_tap($num);
+    }
+
+    return $self->$converter($e, $num);
+}
+
+sub _ok_event {
+    my $self = shift;
+    my ($e, $num) = @_;
+
+    # We use direct hash access for performance. OK events are so common we
+    # need this to be fast.
+    my $name  = $e->{name};
+    my $debug = $e->{debug};
+    my $todo  = $e->{todo};
```

Comment:
Thanks. I do not think the performance is a big factor on this one. That said I do much prefer the my ( $name, $debug, $todo ) = @{$e}{qw( name debug todo)} form, and will likely switch to it because I like it.